spy.go

```diff
@@ -42,3 +103,65 @@ func Execute(w io.Writer, commandLine string) error {
 
 	return nil
 }
+
+// NewSession creates and returns a new shell session with a file
+func NewSession(filepath string, opts ...Option) (session, error) {
+	s := session{}
+	f, err := os.Create(filepath)
+	if err != nil {
+		return session{}, err
+	}
+
+	s.Input = os.Stdin
```

Comment:
We can set all three of these fields in a single assignment by constructing a `session` literal:

```go
s := &session{
	Input: os.Stdin,
	Terminal: os.Stdout,
	Recorder: f,
}
```

(We take the address of it right away, since we'll need a session pointer to pass to the option funcs, and that's what we'll return, too.)